draft-fossati-tls-attestation.md

```diff
@@ -154,74 +154,108 @@ Several formats for encoding evidence are available, such as:
-Certificate message contains a payload that is encoded based on the wrapper defined 
-in {{I-D.ftbs-rats-msg-wrap}}.
-
-In TLS a client has to demonstrate possession of the private key via the CertificateVerify
-message, when client-based authentication is requested. The attestation payload
-must contain a key attestation token, which associates a private key with the
-attestation information. An example of a key attestation token format utilizing 
-the EAT format can be found in {{I-D.bft-rats-kat}}.
-
-The recipient extracts evidence from the Certificate message and relays it to the 
-verifier to obtain attestation results. Subsequently, the attested key is used
-to verify the CertificateVerify message.
-
-# Use of Evidence with the Background Check Model
-
-The background check model is described in Section 5.2 of 
-{{I-D.ietf-rats-architecture}} and allows the following modes
-of operation when used with TLS, namely:
+As typical with new features in TLS, the client indicates support for the new
+extension in the ClientHello message. The newly introduced extensions allow
+remote attestation credentials and nonces to be exchanged. The nonces are used
+for guaranteeing freshness of the exchanged evidence when the background check
+model is in use.
+
+When either the evidence or the results extension is successfully negotiated,
+the content of the corresponding Certificate message contains a payload that is
+encoded based on the wrapper defined in {{I-D.ftbs-rats-msg-wrap}}.
+
+In TLS a client has to demonstrate possession of the private key via the
+CertificateVerify message, when client-based authentication is requested. The
+attestation payload must contain assertions relating to the client's TLS
+Identity Key (TIK), which associate the private key with the attestation
+information. These assertions may come in the form of a Key Attestation Token
+(KAT), or of specific claims in an attestation result document. An example of a
+KAT format utilizing the EAT format can be found in {{I-D.bft-rats-kat}}.
+
+The relying party can obtain and appraise the remote attestation results either
+directly from the Certificate message (in the passport model), or by relaying
+the evidence from the Certificate message to the verifier. Subsequently, the
+attested key is used to verify the CertificateVerify message.
+
+When using the passport model, the remote attestation results obtained by the
+attester from its trusted verifiers can be cached and used for any number of
+subsequent TLS handshakes. This allows the two sides to amortize the cost of
+remote attestation by reducing the frequency with which evidence must be
+produced and appraised. However, this does create a risk of stale attestation
+results being used for authentication. The relying party must therefore be
+empowered to reject an authentication attempt with results that do not meet its
+staleness policy. This should trigger a new evidence appraisal on the attester
+side.
```

Comment:
Suggested edit:
When using the passport model, the remote attestation results obtained by the
attester from its trusted verifiers can be cached and used for any number of
subsequent TLS handshakes. This allows the two sides to amortize the cost of
remote attestation by reducing the frequency with which remote attestation needs to be
performed with the verifier.  However, this does create a risk of stale attestation
results being used for authentication. The relying party must therefore be
empowered to reject an authentication attempt with results that do not meet its
acceptance policy. This should trigger a new attestation protocol between attester 
and the verifier.

Reply:
Also a generic question, we should specify 
(a) how existing protocols built in the TLS system allow us to reject the Attestation results

(b) If a suitable cause needs to be added in the Reject Message, what the new value should be?